packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -179,6 +180,42 @@ export interface CanaryProps {
    * @default - No environment variables.
    */
   readonly environmentVariables?: { [key: string]: string };
+
+  /**
+   * The VPC where this canary is run.
+   *
+   * Specify this if the canary needs to access resources in a VPC.
+   *
+   * @default - Not in VPC
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * Where to place the network interfaces within the VPC. You must provide `vpc` when using this prop.
+   *
+   * @default - the Vpc default strategy if not specified
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+
+  /**
+   * Whether or not to allow placing a canary to be run in only private subnets. By default, the Synthetics runtime
+   * needs access to the S3 and CloudWatch APIs, which will fail in a private subnet without internet access enabled.
+   *
+   * You can bypass this check if you've enabled VPC Endpoints for these two APIs in your VPC or have an Internet Gateway setup.
+   *
+   *  You must provide `vpc` when using this prop.
+   *
+   * @default false
+   */
+  readonly allowPrivateSubnet?: boolean;
```

Comment:
I don't think this is needed. I think it should be fine to just add this to the documentation in the readme and the `vpcSubnets` docstring. Users will specify if they want to place it in isolated subnets via the `vpcSubnets` prop, asking again here feels like a "are you sure you know what you are doing?" and I don't think that is needed.